### PR TITLE
Webpack + SWC Loader

### PR DESCRIPTION
**Pull request by joel-jeremy**
Related to https://github.com/actualbudget/actual/issues/656

### CODE REVIEW

**Comment by joel-jeremy** [comment]
Current build time: `time yarn build:browser`
image

SWC build time: `time yarn build:browser`
image

**Comment by MatissJanis** [comment]
Really cool. `react-scripts` was deprecated ages ago. It's nice to see us moving to something more modern.

Let me know when you want me to review this.

**Comment by joel-jeremy** [comment]
# loot-core tests: (-44% in execution time)

## ts-jest execution time
image

## SWC execution time
image

# crdt tests (-88.2% in execution time)

## ts-jest execution time
image

## SWC execution time
image

For some reason crdt tests fail in local but succeeds in CI. I haven't looked into why that is.

**Comment by joel-jeremy** [comment]
@MatissJanis @shall0pass @trevdor This is ready for review :) There seem to be an increase in bundle size due to the switch but build and test execution time has dramatically improved.

Aside: It looks like github action is not caching node modules?

File: packages/desktop-client/package.json [hunk 1]
```diff
@@ -40,7 +43,6 @@
     "memoize-one": "^6.0.0",
     "pikaday": "1.8.0",
     "react": "18.2.0",
-    "react-app-rewired": "^2.2.1",
```
Comment (MatissJanis) [inline packages/desktop-client/package.json:43]: **💬 suggestion:** ‏`react-scripts` can also be removed IMO

**Comment by MatissJanis** [comment]
Just pushed a small patch for electron app I found while testing this PR. Hope you don't mind.

File: .github/actions/setup/action.yml [hunk 2]
```diff
@@ -7,12 +7,7 @@ runs:
       uses: actions/setup-node@v3
       with:
         node-version: 18.16.0
-    - name: Cache
-      uses: actions/cache@v3
-      id: cache
-      with:
-        path: '**/node_modules'
-        key: yarn-v1-${{ runner.os }}-${{ hashFiles('.nvmrc') }}-${{ hashFiles('**/yarn.lock') }}
+        cache: yarn
```
Comment (MatissJanis) [inline .github/actions/setup/action.yml:10]: **💬 suggestion:** ‏The default cache mechanism does not take into account `.nvmrc` file. That's why we created a custom cache key here.

With the proposed change the cache will not be invalidated when we bump the node version in `.nvmrc`

**Review by MatissJanis — COMMENTED** [review]
Did not spot any significant issue! Amazing job here :)

**Comment by MatissJanis** [comment]
> Just pushed a small patch for electron app I found while testing this PR. Hope you don't mind.

Ok, this broke VRT.. apologies. I reverted my change. Will patch it in a separate PR.

**Comment by MatissJanis** [comment]
🤦 it was not my changes.. the VRT in `master` are running against your PR. And this PR doesn't have the change you merged a few minutes ago.

Updating the branch should solve the issue

File: packages/desktop-client/package.json (hunk 1 above)
Comment (joel-jeremy) [reply packages/desktop-client/package.json:43]: Craco still uses react-scripts under the hood I believe: https://craco.js.org/docs/getting-started/

File: .github/actions/setup/action.yml (hunk 2 above)
Comment (joel-jeremy) [reply .github/actions/setup/action.yml:10]: Oh I see, I'll revert this change

**Comment by joel-jeremy** [comment]
I have rebased the branch. Feel free to commit your electron fix :)

**Review by MatissJanis — APPROVED** [review]

**Pull request merged**